core/src/main/scala/org/http4s/blaze/pipeline/Stages.scala

```diff
@@ -255,12 +244,18 @@ sealed trait Head[O] extends Stage {
   /** Send a command to the next inbound `Stage` of the pipeline */
   final def sendInboundCommand(cmd: InboundCommand): Unit = {
     logger.debug(s"Stage ${getClass.getSimpleName} sending inbound command: $cmd")
-    if (_nextStage != null) {
-      try _nextStage.inboundCommand(cmd)
-      catch { case t: Throwable => outboundCommand(Error(t)) }
+    val next = _nextStage
+    if (next != null) {
+      try next.inboundCommand(cmd)
+      catch {
+        case t: Throwable =>
+          logger.error(t)("Exception caught when attempting inbound command")
+          closePipeline(Some(t))
+      }
     } else {
-      val e = new Exception("Cannot send inbound command on disconnected stage")
-      logger.error(e)("")
+      val msg = "Cannot send inbound command on disconnected stage"
+      val e = new Exception(msg)
+      logger.error(e)(msg)
```

Comment:
Not new, but why are we logging and throwing the same exception here?  I guess it's the age old discussion of the ideal being exactly once logging and the counterargument of at-least once logging.

Reply:
Old code since before we had that discussion. I think we should remove the log and just throw the exception. I'll change it.